lib/services/ios-project-service.ts

```diff
@@ -421,11 +421,16 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
 	}
 
 	private async setupSigningForDevice(projectRoot: string, buildConfig: IiOSBuildConfig, projectData: IProjectData): Promise<void> {
-		const pbxprojPath = path.join(projectRoot, projectData.projectName + ".xcodeproj", "project.pbxproj");
-		const xcode = Xcode.open(pbxprojPath);
+		const xcode = Xcode.open(this.getPbxProjPath(projectData));
 		const signing = xcode.getSigning(projectData.projectName);
 
-		if ((this.readXCConfigProvisioningProfile(projectData) || this.readXCConfigProvisioningProfileForIPhoneOs(projectData)) && (!signing || signing.style !== "Manual")) {
+		const hasProvisioningProfileInXCConfig =
+			this.readXCConfigProvisioningProfile(projectData) ||
```

Comment:
I think the ordering here should be the other way around `[sdk=iphoneos*]` lines in .xcconfig are more specific than the ones without and it makes more sense that specific ones are taken into account before general ones.
In addition, `PROVISIONING_PROFILE_SPECIFIER` is a newer term than `PROVISIONING_PROFILE` and the latter is close to being deprecated I think.
So I think the order should be:
```TypeScript
this.readXCConfigProvisioningProfileSpecifierForIPhoneOs(projectData) ||
this.readXCConfigProvisioningProfileSpecifier(projectData) ||
this.readXCConfigProvisioningProfileForIPhoneOs(projectData) ||
this.readXCConfigProvisioningProfile(projectData)
```